types/shared.ts

```diff
@@ -0,0 +1,31 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+import { CorrelationFinding } from "./Correlations";
+import { DetectorHit } from "./Detector";
+import { Finding } from "./Finding";
+
+export interface AbortSignal {
+  signal: boolean;
```

Comment:
also, I think signal to a `boolean` variable doesn't sound a good name to me, I think we can rename it to `aborted`? or `isAborted` or `cancelled`?  it will give more clarity that way.

Reply:
Refactored to use the inbuilt controller